data/data/openstack/masters/main.tf

```diff
@@ -63,7 +63,7 @@ resource "openstack_compute_instance_v2" "master_conf_0" {
 
   flavor_id = data.openstack_compute_flavor_v2.masters_flavor.id
   image_id = var.root_volume_size == null ? var.base_image_id : null
-  security_groups = var.master_sg_ids
+  security_groups = var.master_sg_ids == null ? [] : var.master_sg_ids
```

Comment:
This should not be needed. (1) The resource should be able to handle null. (2) The variable will never be null: It is an empty list when security groups are disabled.